thetis/options.py

```diff
@@ -29,6 +29,8 @@ class SemiImplicitTimestepperOptions2d(TimeStepperOptions):
         'ksp_type': 'gmres',
         'pc_type': 'sor',
     }).tag(config=True)
+    use_semi_implicit_linearization = Bool(
```

Comment:
Also, to confirm I understand: the default is now changed to solve the full nonlinear system, right? I think this would be a reasonable (robust) default indeed...

Reply:
Ah, that's right: `CrankNicolsonTimestepperOptions2D` is inherited from the `SemiImplicitTimestepperOptions2d` class. I'll fix that.
Yes, in the 2D model the default behavior is now `use_semi_implicit_linearization=True` for all implicit schemes, which makes more sense.